noria-server/mir/src/optimize.rs

```diff
@@ -1,19 +1,217 @@
+                                use nom_sql::FunctionExpression::{Count, Sum};
+                                for (i, col) in child.columns.iter().enumerate() {
+                                    match col.function.clone() {
+                                        Some(func_expr) => {
+                                            match *func_expr {
+                                                 Count(col, _)
+                                                 | Sum(col, _) => {
+                                                    if col.name == on.name {
+                                                        // this column may be the aggregation result
+                                                        // so if we're filtering on it, we're not a candidate
+                                                        for (j, _cond) in conditions {
+                                                            if *j == i {
+                                                                candidate = false;
+                                                            }
+                                                        }
+                                                    }
+                                                },
+                                                _ => {},
+                                            }
+                                        },
+                                        None => {},
+                                    }
+                                }
+                            }
+                        },
+                        _ => {},
+                    };
+                }
+            },
+            _ => {},
+        };
+        if candidate {
+            candidate_nodes.push(n);
+        }
+    }
+
+    // 3. For each candidate, merge it, and update all parents/children
+    // of the newly merged node.
+
+    let mut new_nodes = Vec::new();
+
+    for n in candidate_nodes {
+        let temp = n.borrow();
+        let child = temp.children.first().unwrap().borrow();
+        // determine which is which
+        // (this was relevant when we supported either order of agg/filter,
+        // but I'm leaving it in place in case we resume that)
+        let (agg, filter) = match n.borrow().inner {
+            MirNodeType::Aggregation { .. }  => (n.borrow().inner.clone(), child.inner.clone()),
+            MirNodeType::Filter { .. } => (child.inner.clone(), n.borrow().inner.clone()),
+            _ => unreachable!(),
+        };
+
+        let mut new_name = child.name.clone();
+        new_name.push_str("_filteragg");
+
+        let new_node = MirNode::new(
+            &new_name,
+            child.from_version,
+            child.columns.clone(),
+            MirNodeType::FilterAggregation {
+                on: match &agg {
+                    MirNodeType::Aggregation { on, group_by: _, kind: _ } => on.clone(),
```

Comment:
You can just use `{ on, .. }` here :)

Reply:
So, in this case I did know there was a shorter way to write it, but I personally found it more readable/modifiable to see what all the fields are and which ones we're using here. Why do you prefer the other way?

I'll defer to your preferred style here, of course. I'm just curious for your reasoning.

Reply:
Ah, so, the reason is in case we later add _more_ fields to `Aggregation`. This pattern would then fail to compile, whereas one with `, ..` would not. Although that could also be considered a feature, depending on how sensitive you think this could would be to `Aggregation` changing.